### PR TITLE
Validate bulk uploader device info

### PR DESCRIPTION
**Pull request by zdeveloper**
# BACKEND PULL REQUEST

## Related Issue

- #4889

## Changes Proposed

- validate device model + test performed code combo against DeviceType table
- validate test performed code against flu only list

## Additional Information

- this will be merged after the device sync https://github.com/CDCgov/prime-simplereport/pull/5491
- Validation logic https://github.com/CDCgov/prime-simplereport/wiki/SR-Result-bulk-uploader-device-validation-logic

## Testing

- How should reviewers verify this PR?

### CODE REVIEW

**Review by BobanL — DISMISSED** [review]
Code changes look good! Tested in dev3 and works as expected.

**Review by DanielSass — DISMISSED** [review]

**Comment by lg-king** [comment]
@zdeveloper CA confirmed that they would like us to require the `equipment_model_name` field for flu-only devices: 

> Yes, we’d like to keep the device name field required, even though validation on the name/LOINC combo may not be perfect.

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/filerow/TestResultRow.java [hunk 1]
```diff
@@ -122,6 +126,92 @@ public class TestResultRow implements FileRow {
   static final String TESTING_LAB_CITY = "testing_lab_city";
   static final String TESTING_LAB_STATE = "testing_lab_state";
   static final String TESTING_LAB_ZIP_CODE_FIELD = "testing_lab_zip_code";
+  static Set<String> fluOnlyTestPerformedLoinc =
+      Set.of(
+          "100973-7",
```
Comment (nathancrtr) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/filerow/TestResultRow.java:131]: Are these LOINCs ever represented in documentation or guidance without the dash `-`? Wondering if we would ever want to accept LOINCs in a different format.

Comment (zdeveloper) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/filerow/TestResultRow.java:131]: Great observation, the idea was to add the validation in place and then update the docs once we are ready to release it

**Review by BobanL — APPROVED** [review]
Changes look good and tested succesfully in dev6! 

The only thing I noticed was that default example file does throw an error, I believe it's just the [model names need to have the `*` removed](https://github.com/CDCgov/prime-simplereport-site/blob/4fb6a213f50206bc13283bfda547f4e05f034896/assets/resources/test_results_example_10-3-2022.csv#L1:~:text=BD%20Veritor%20System%20for%20Rapid%20Detection%20of%20SARS%2DCoV%2D2%20%26%20Flu%20A%2BB*). 
image

**Comment by zdeveloper** [comment]
> Changes look good and tested succesfully in dev6!
> 
> The only thing I noticed was that default example file does throw an error, I believe it's just the [model names need to have the `*` removed](https://github.com/CDCgov/prime-simplereport-site/blob/4fb6a213f50206bc13283bfda547f4e05f034896/assets/resources/test_results_example_10-3-2022.csv#L1:~:text=BD%20Veritor%20System%20for%20Rapid%20Detection%20of%20SARS%2DCoV%2D2%20%26%20Flu%20A%2BB*). image

Great catch, we recently removed all the `*` when we synced with the livd table, will update the example file

**Comment by zdeveloper** [comment]
[PR](https://github.com/CDCgov/prime-simplereport-site/pull/502) to update the results bulk uploader example csv file

File: backend/src/main/java/gov/cdc/usds/simplereport/service/ResultsUploaderDeviceValidationService.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ResultsUploaderDeviceValidationService {
+  private final DeviceTypeRepository deviceTypeRepository;
+
+  @Cacheable(DEVICE_MODEL_AND_TEST_PERFORMED_CODE_SET)
```
Comment (nathancrtr) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/ResultsUploaderDeviceValidationService.java:23]: Cool stuff here! I'll have a little reading to do on some of the implementation details but this is a nice addition.

**Review by nathancrtr — APPROVED** [review]
Tested and this LGTM! Good catch on the doc update on the static site.

**Pull request merged**